__tests__/app.test.js

```diff
@@ -285,9 +286,48 @@ describe('PATCH:', () => {
 		});
 	});
 });
-
-
-
-//You could add another test for an invalid_id, api/articles/banana/comments etc. Should be a 400.
-
-//Also are there any articles that exist but don't have comments? If so you should return an empty array!
\ No newline at end of file
+describe('POST:', () => { 
+	describe('/api/articles/:article_id/comments', () => { 
+		test('status:202 assigns new comment to database from a known user and returns comment', () => {
+			return request(app)
+				.post('/api/articles/1/comments')
+				.send({
+					username: 'butter_bridge',
+					body: 'test comment'
+				})
+				.expect(202)
+				.then(res => {
+					expect(res._body).toEqual(
+						expect.objectContaining({
+							article_id: 1,
+							author: 'butter_bridge',
+							body: 'test comment',
+							comment_id: 19,
+							created_at: expect.any(String),
+							votes: 0
+						})
+					)
+				});
+		 })
+		test('status:404 returns a error when the user dose not exist', () => {
+			return request(app)
+				.post('/api/articles/1/comments')
+				.send({
+					username: 'unknown_user',
+					body: 'test comment'
+				})
+				.expect(404)
+				.then(res => {expect(res._body).toEqual({ msg: 'Not found' })});
+		 })
+		test('status:400 rejects a comment without a valid comment', () => {
+			return request(app)
+				.post('/api/articles/1/comments')
+				.send({
+					username: 'butter_bridge',
+					body: 8
+				})
+				.expect(400)
+				.then(res => { expect(res._body).toEqual({ msg: "Bad request" }) })
+		 })
+	})
+})
```

Comment:
lovely tests, please can you add the usual  - non-existent id and invalid id too